tests/javascripts/timeout.test.js

```diff
@@ -0,0 +1,206 @@
+      expect(isDialogOpen(inactivity_popup)).toEqual(true); // displays after period of inactivity
+    } else {
+      expect(isDialogOpen(inactivity_popup)).toEqual(false);
+    }
+  });
+
+  test("Inactivity dialog text appears in dialog", () => {
+    jest.advanceTimersByTime(60 * inactivity_period * 1000);
+
+    if (window.GOVUK.isLoggedIn()) {
+      expect(isDialogOpen(inactivity_popup)).toEqual(true);
+      expect(inactivity_popup.querySelector("h1").textContent).toBe(
+        "You're about to be signed out"
+      );
+      expect(inactivity_popup.querySelector("p").textContent).toBe(
+        "For your security, we will sign you out in 2 minutes."
+      );
+    } else {
+      expect(isDialogOpen(inactivity_popup)).toEqual(false);
+    }
+  });
+
+  test("Stay signed in button should close dialog and reset timeout ", () => {
+    jest.advanceTimersByTime(60 * inactivity_period * 1000);
+    if (window.GOVUK.isLoggedIn()) {
+      expect(isDialogOpen(inactivity_popup)).toEqual(true);
+
+      helpers.triggerEvent(inactivity_button, "click");
+
+      expect(isDialogOpen(inactivity_popup)).toEqual(false);
+    } else {
+      expect(isDialogOpen(inactivity_popup)).toEqual(false);
+    }
+  });
+
+  test("Sign out button should close dialog and take to sign out screen", () => {
+    const link = document.getElementById("hmrc-timeout-sign-out-link");
+    const hasBeenClicked = jest.fn();
+    link.addEventListener("click", hasBeenClicked);
+
+    jest.advanceTimersByTime(60 * inactivity_period * 1000);
+    if (window.GOVUK.isLoggedIn()) {
+      expect(isDialogOpen(inactivity_popup)).toEqual(true);
+
+      helpers.triggerEvent(link, "click");
+
+      expect(hasBeenClicked).toHaveBeenCalled();
+    } else {
+      expect(isDialogOpen(inactivity_popup)).toEqual(false);
+    }
+  });
+});
+
+describe("Timeout dialog appears", () => {
+  beforeEach(() => {
+    document.body.innerHTML = html_content;
+    jest.useFakeTimers();
+    window.GOVUK.sessionExpiryPopup();
+  });
+
+  afterEach(() => {
+    document.body.innerHTML = "";
+  });
+
+  test("Dialog opens after timeout", () => {
+    const expiry_popup = document.querySelector("#expiry");
+    jest.advanceTimersByTime(60 * timeout_period * 1000);
+    if (window.GOVUK.isLoggedIn()) {
+      expect(isDialogOpen(expiry_popup)).toEqual(true);
+
+      helpers.triggerEvent(link, "click");
+
+      expect(hasBeenClicked).toHaveBeenCalled();
+    } else {
+      expect(isDialogOpen(expiry_popup)).toEqual(false);
+    }
+  });
+
+  test("Expiry dialog text appears in dialog", () => {
+    if (window.GOVUK.isLoggedIn()) {
+      expect(expiry_popup.querySelector("h1").textContent).toBe(
+        "You can no longer extend your session"
+      );
+    } else {
+      expect(isDialogOpen(expiry_popup)).toEqual(false);
+      // not to be visible
+    }
+  });
+
+  test("Sign in again button should close dialog and sign out ", () => {
+    const expiry_button = document.getElementById(
```

Comment:
Again we seem to be using getElementById and getSelector (with an id passed) interchangeably, we should pick one and stick to it

Reply:
I've removed all references to querySelector in favour of getElementbyId or getElementsByTagName for the selection of certain tags.